### PR TITLE
feat(Composer): Add overlay for matterport tags

### PR DESCRIPTION
**Pull request by mukeshsahay**
## Overview
- Add data overlay for matterport tags with label and description (match with Matterport portal).
- Enabled overlay feature flag for Matterport story

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/scene-composer/src/hooks/useMatterportTags.ts [hunk 1]
```diff
@@ -2,9 +2,32 @@ import { useCallback } from 'react';
...
-import { IAnchorComponentInternal, ISceneNodeInternal, useStore } from '../store';
+import { IAnchorComponentInternal, IDataOverlayComponentInternal, ISceneNodeInternal, useStore } from '../store';
 import { MattertagItem, TagItem } from '../utils/matterportTagUtils';
 import { RecursivePartial } from '../utils/typeUtils';
+import { Component } from '../models/SceneModels';
+import { generateUUID } from '../utils/mathUtils';
+
+const getContentForOverlayComponent = (label: string, description: string): string => {
+  return `#### **${label}**  
+${description}`;
```
Comment (sheilaXu) [inline packages/scene-composer/src/hooks/useMatterportTags.ts:13]: this indentation looks weird.. can we just add `\n` to make new line?

Comment (mukeshsahay) [reply packages/scene-composer/src/hooks/useMatterportTags.ts:13]: This is the preferred way from Markdown guideline.

File: packages/scene-composer/src/hooks/useMatterportTags.ts [hunk 2]
```diff
@@ -27,6 +50,9 @@ const addTag = (
       matterportId: id, //mattertag uses item.sid and tag uses item.id so we just us the collection key for both
     },
   } as ISceneNode;
+
+  const dataoverlayComponent = getNewDataOverlayComponent(item);
+  node.components?.push(dataoverlayComponent);
```
Comment (sheilaXu) [inline packages/scene-composer/src/hooks/useMatterportTags.ts:55]: can we create the overlay before creating the node and directly adding to the components at line43? having this part separately will easily get missed

File: packages/scene-composer/src/hooks/useMatterportTags.ts (hunk 1 above)
Comment (sheilaXu) [reply packages/scene-composer/src/hooks/useMatterportTags.ts:13]: also is that possible for the item to not have label and/or description? so we may only have one line or even no overlay at all?

Comment (mukeshsahay) [reply packages/scene-composer/src/hooks/useMatterportTags.ts:13]: Label will always be there.

Comment (mukeshsahay) [reply packages/scene-composer/src/hooks/useMatterportTags.ts:13]: Label is similar to the Tag name.

File: packages/scene-composer/src/hooks/useMatterportTags.ts (hunk 2 above)
Comment (mukeshsahay) [reply packages/scene-composer/src/hooks/useMatterportTags.ts]: Done

File: packages/scene-composer/src/hooks/useMatterportTags.ts (hunk 1 above)
Comment (mukeshsahay) [reply packages/scene-composer/src/hooks/useMatterportTags.ts:14]: Added a comment in code

**Review by sheilaXu — APPROVED** [review]

**Pull request merged**